### PR TITLE
Feature / Communities and community page

### PR DESCRIPTION
**Pull request by hugomarquesdev**
- Request Page and Community Page are merged into one page only: `Community Page`
- Filter Requests by Status too (only show requests with status: pending)
- Communities Page
- Requests Page divided in components
- Add manager in community page (at least the design part)

### CODE REVIEW

File: customtypes/pwa-view-proposals/index.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "id": "pwa-view-proposals",
+  "label": "[PWA] [View] Proposals",
+  "repeatable": false,
+  "status": true,
+  "json": {
+    "Main": {}
+  }
+}
```
Comment (psoaresbj) [inline customtypes/pwa-view-proposals/index.json:9]: nl

File: src/utils/routes.ts [hunk 2]
```diff
@@ -2,10 +2,10 @@ export type Routes = string[];
...
-export const privateRoutes: Routes = ['/profile', '/settings', '/requests', `/requests/[id]`];
+export const privateRoutes: Routes = ['/profile', '/settings'];
 
 export const demoRoutes: Routes = [];
 
-export const beneficiaryRoutes: Routes = ['/beneficiary', '/requests', `/requests/[id]`];
+export const beneficiaryRoutes: Routes = ['/beneficiary', '/requests', '/communities', '/communities/[id]'];
 
-export const managerRoutes: Routes = [];
+export const managerRoutes: Routes = ['/communities', '/communities/[id]'];
```
Comment (psoaresbj) [inline src/utils/routes.ts:11]: nl

File: src/views/Communities.tsx [hunk 3]
```diff
@@ -16,41 +16,35 @@ import {
...
-const Requests: React.FC<{ isLoading?: boolean }> = (props) => {
+const Communities: React.FC<{ isLoading?: boolean }> = (props) => {
     const { isLoading } = props;
 
-    const { extractFromView } = usePrismicData();
-    const { title, content } = extractFromView('heading') as any;
-
-    //  Review has 4 states: 'pending', 'accepted', 'claimed', 'declined'
+    // const { extractFromView } = usePrismicData();
+    // const { title, content } = extractFromView('heading') as any;
```
Comment (psoaresbj) [inline src/views/Communities.tsx:31]: 👀

File: src/views/Communities.tsx [hunk 4]
```diff
@@ -16,41 +16,35 @@ import {
     Text,
     ViewContainer
 } from '@impact-market/ui';
+import { getImage } from '../utils/images';
 
 import { useGetCommunitiesMutation } from '../api/community';
-import { useGetUserMutation } from '../api/user';
 
-import { usePrismicData } from '../libs/Prismic/components/PrismicDataProvider';
+// import { usePrismicData } from '../libs/Prismic/components/PrismicDataProvider';
```
Comment (psoaresbj) [inline src/views/Communities.tsx:23]: 👀

File: src/views/Communities.tsx [hunk 5]
```diff
@@ -64,42 +58,37 @@ const Requests: React.FC<{ isLoading?: boolean }> = (props) => {
...
 
     return (
         <ViewContainer isLoading={isLoading}>
             <Display g900 medium>
-                {title}
+                {/* {title} */}
+                My Communities
             </Display>
-            <RichText content={content} g500 mt={0.25} />
+            {/* <RichText content={content} g500 mt={0.25} /> */}
```
Comment (psoaresbj) [inline src/views/Communities.tsx:78]: 👀

File: src/views/Community/Beneficiaries.tsx [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+        <Grid cols={{ sm: 4, xs: 2 }}>
+            <Card>
+                <Text g500 mb={0.3} regular small>
+                    # <String id="beneficiaries"/>
+                </Text>
+                <Grid cols={2}>
+                    <Text g900 medium semibold>
+                        {community?.state &&
+                            !!Object.keys(community).length ? community?.state?.beneficiaries : '-'
+                        }
```
Comment (psoaresbj) [inline src/views/Community/Beneficiaries.tsx:25]:
```suggestion
                        {!!community?.state &&
                            !!Object.keys(community).length ? community?.state?.beneficiaries : '-'
                        }
```

File: src/views/Community/Beneficiaries.tsx [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+                    </Box>
+                </Grid>
+            </Card>
+            <Card>
+                <Text g500 mb={0.3} regular small>
+                    <String id="claimPerBeneficiary"/>
+                </Text>
+                <Grid cols={2}>
+                    <Text g900 medium semibold>
+                        {community?.state && !!Object.keys(community).length ? community?.state?.claims : '-'}
```
Comment (psoaresbj) [inline src/views/Community/Beneficiaries.tsx:44]:
```suggestion
                        {!!community?.state && !!Object.keys(community).length ? community?.state?.claims : '-'}
```

File: src/views/Community/Beneficiaries.tsx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+                            size={2}
+                        />
+                    </Box>
+                </Grid>
+                
+            </Card>
+            <Card>
+                <Text g500 mb={0.3} regular small>
+                    {/* <String id="maximumPerBeneficiary"/> */}
+                    {/* ADD ON PRISMIC */}
```
Comment (psoaresbj) [inline src/views/Community/Beneficiaries.tsx:61]: 👀

File: src/views/Community/Beneficiaries.tsx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+                            s600
+                            size={2}
+                        />
+                    </Box>
+                </Grid>
+                
+            </Card>
+            <Card>
+                <Text g500 mb={0.3} regular small>
+                    {/* ADD ON PRISMIC */}
```
Comment (psoaresbj) [inline src/views/Community/Beneficiaries.tsx:101]: 👀

File: src/views/Community/Header.tsx [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+                            <Box inlineFlex> 
+                                {/* Todo: edit details */}
+                                <Button>
+                                    <Icon
+                                        icon="edit"
+                                        margin="0 0.5 0 0"
+                                        n01
+                                    />
+                                    {/* <String id="editDetails"/> */}
+                                    {/* ADD ON PRISMIC */}
```
Comment (psoaresbj) [inline src/views/Community/Header.tsx:78]: 👀

File: src/views/Community/Header.tsx [hunk 11]
```diff
@@ -0,0 +1,126 @@
...
+                                {community?.review === 'accepted' ? (
+                                    <>
+                                        <Button ml={0.5} secondary>
+                                            <Icon
+                                                icon="share"
+                                                margin="0 0.5 0 0"
+                                                p700
+                                            />
+                                            {/* <String id="editDetails"/> */}
+                                            {/* ADD ON PRISMIC */}
```
Comment (psoaresbj) [inline src/views/Community/Header.tsx:91]: 👀

File: src/views/Community/Managers.tsx [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+        {
+            avatar:'https://picsum.photos/120',
+            beneficiaries:'Onboarded 22 Beneficiaires.',
+            id:'0x43D2...a12f7',
+            mail:'johndoe@myemail.com',
+            manager:'Manager from May 12, 2020 – Present',
+            name:'Paulo Silva Queirós',
+            phone:'+1 669 2319892'   
+        }
+    ]
```
Comment (psoaresbj) [inline src/views/Community/Managers.tsx:49]: If you want to approve this and want to add mock data, create a separate file where we return a promise wrapped with a timeout and create the request as we should be fetching data from API.

File: src/views/Community/index.tsx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+            //  Todo: text for toasters
+            toast.success('Successfully changed community review state!');
+
+            //  Send to /requests if community was declined
+            review === 'declined' && router.push('/requests')
+
+        } catch (error) {
+            console.log(error);
+
+            //  Todo: text for toasters
```
Comment (psoaresbj) [inline src/views/Community/index.tsx:54]: 👀

File: src/views/Community/index.tsx [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+                id: communityId
+            });
+
+            const community: any = await getCommunity(communityId);
+
+            setCommunity(community.data);
+
+            setLoading(false);
+
+            //  Todo: text for toasters
```
Comment (psoaresbj) [inline src/views/Community/index.tsx:45]: 👀

File: src/views/Requests/ReviewTabs.tsx [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+            )
+        :
+            !!Object.keys(numberOfCommunitiesByReview).length && (
+                numberOfCommunitiesByReview.filter((reviewName: any) => reviewName.review === review).map((count: any) => (
+                    otherCountriesNumberOfReviews = parseInt(count.count, 10)
+                ))
+            )
+
+        return !!myCountrySelected ? parseInt(myCountryNumberOfReviews[review], 10) : otherCountriesNumberOfReviews
+    }
```
Comment (psoaresbj) [inline src/views/Requests/ReviewTabs.tsx:38]: I don't understand this function.

When the country is selected you set the last position of the filtered results, as `myCountryNumberOfReviews` but then you treat it as an obj or arr?

Comment (hugomarquesdev) [reply src/views/Requests/ReviewTabs.tsx:38]: `reviewsByCountry` returns objects with all countries and review status of each one, like so:

```
accepted: "1"
claimed: "0"
count: "1"
country: "BR"
declined: "0"
pending: "0"

accepted: "1"
claimed: "0"
count: "1"
country: "PT"
declined: "0"
pending: "0"
```
etc...

I run a .filter in `reviewsByCountry` to return only the object of user's country and assign it to `myCountryNumberOfReviews`.

At the end of the function I return the value of the key `[review]` (for example, pending) from the `myCountryNumberOfReviews` object, and transform it to a number like so: `parseInt(myCountryNumberOfReviews[review], 10)`

**Review by psoaresbj — APPROVED** [review]

**Pull request merged**